app/src/main/java/covid/trace/morocco/LocaleHelper.kt

```diff
@@ -0,0 +1,92 @@
+package covid.trace.morocco
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringDef
+import io.github.inflationx.calligraphy3.CalligraphyConfig
+import io.github.inflationx.calligraphy3.CalligraphyInterceptor
+import io.github.inflationx.viewpump.ViewPump
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.util.*
+
+object LocaleHelper {
+
+    private const val SHARED_PREF_NAME = "covid_19"
+    private const val PATH_FONT_FR = "fonts/Barlow-Regular.ttf"
+    private const val PATH_FONT_AR = "fonts/Almarai-Regular.ttf"
+    private const val LANGUAGE_PREF_KEY = "language_pref_key"
+    const val FRENCH = "fr"
+    const val ARABIC = "ar"
+
+    private var sharedPreferences: SharedPreferences? = null
```

Comment:
a `lateinit` `sharedPreferences` should be more appropriate. don't you think?

Reply:
sharedPreferences can be null if a class calls a method (for example the method setLocaleIfNeeded(@Language language: String)) without initiating sharedPreferences via the method init(context: Context?)

Reply:
OK I see BUT calling `init` is mandatory. It's a prerequisite to use `LocaleHelper`. As you can see init is called here https://github.com/Wiqaytna-app/wiqaytna_android/pull/12/files#diff-72d398151f1a0977873d51d832bb38f2R33